libraries/eosiolib/core/eosio/crypto_bls_ext.hpp

```diff
@@ -0,0 +1,103 @@
+/**
+ *  @file
+ *  @copyright defined in cdt/LICENSE
+ */
+#pragma once
+
+#include "fixed_bytes.hpp"
+#include "varint.hpp"
+#include "serialize.hpp"
+
+#include <array>
+
+namespace eosio {
+
+
+   //using bls_public_key = std::array<char, 48>;
+   //using bls_signature = std::array<char, 96>;
+
+
+   using bls_public_key = std::array<uint8_t, 49>;
+   using bls_signature = std::array<uint8_t, 97>;
+
+   namespace internal_use_do_not_use {
+      extern "C" {
+
+         __attribute__((eosio_wasm_import))
+         bool bls_verify( const char* sig, uint32_t sig_len, const char* dig, uint32_t dig_len, char* pub, uint32_t pub_len);
+
+         __attribute__((eosio_wasm_import))
+         int32_t bls_aggregate_pubkeys( const char* pubs, uint32_t pubs_len, const char* agg, uint32_t agg_len);
+
+         __attribute__((eosio_wasm_import))
+         int32_t bls_aggregate_sigs( const char* sigs, uint32_t sigs_len, const char* agg, uint32_t agg_len);
+
+         __attribute__((eosio_wasm_import))
+         bool bls_aggregate_verify( const char* sig, uint32_t sig_len, const char* digs, uint32_t digs_len, char* pubs, uint32_t pubs_len);
+
+      }
+
+   }
+
+   bool bls_verify(bls_public_key pub, bls_signature sig , std::vector<uint8_t> msg) {

```

Comment:
pass by reference or explain why do you pass parameters by value here. same for other wrappers